src/lib/eina/eina_error.c

```diff
@@ -350,7 +350,7 @@ eina_error_msg_get(Eina_Error error)
                EINA_SAFETY_ERROR("strerror_r() failed");
              else
                {
-                  if (strncmp(str, unknown_prefix, sizeof(unknown_prefix) -1) == 0)
+                  if (strncmp(str, unknown_prefix, sizeof(unknown_prefix) - 2) == 0)
```

Comment:
For me it makes more sense to actually change the expected error message (since `"Unknown error"` is more a prefix than `"Unknown error "`).

Reply:
Also, I would proposa an `is_prefix` function, so you can compare if `unknown_prefix` is a prefix of `str` (i.e. if `"Unknown error"` is prefix of `"Unknown error 1234"`).

Reply:
why would you need a function? Isn't this in a test?

Reply:
This is in `src/lib/eina/eina_error.c`, not in `src/tests`. Besides, seeing `is_prefix` instead of an inline for-loop seems more reasonable/readable for me.

Reply:
We need to look into the goal of a diff. Doing more than what the commit is meant to do is not good. If we want a function that does that, that could be another commit. One that is not high priority for us.

Reply:
A good commit does the least amount of modifications needed to accomplish its goal.